zerver/lib/test_classes.py

```diff
@@ -1152,6 +1153,23 @@ def get_streams(self, user_profile: UserProfile) -> list[str]:
             ).values_list("name", flat=True)
         )
 
+    def get_dm_recipient(self, sender: UserProfile, *other_users: UserProfile) -> Recipient:
+        if settings.PREFER_DIRECT_MESSAGE_GROUP or len(other_users) > 1:
+            direct_group_message = get_or_create_direct_message_group(
+                id_list=[sender.id] + [user.id for user in other_users],
+            )
+            assert direct_group_message.recipient is not None
+            return direct_group_message.recipient
+        else:
+            assert len(other_users) <= 1
+            if len(other_users) == 1:
+                assert other_users[0].recipient is not None
+                return other_users[0].recipient
+            else:
+                # This is the case where we are sending a message to ourselves.
+                assert sender.recipient is not None
+                return sender.recipient
```

Comment:
Not sure if having a helper function like this complicates unit tests, since it abstracts away the recipient type from the test function. But without it, we’d have to duplicate a lot of tests (one with `PREFER_DIRECT_MESSAGE_GROUP` enabled and one with it disabled).